desktop_app/src/backend/server/plugins/mcp/index.ts

```diff
@@ -85,21 +232,126 @@ export const createArchestraMcpServer = () => {
   );
 
   archestraMcpServer.tool(
-    'uninstall_mcp_server',
+    'delete_memory',
+    'Delete a specific memory entry by name',
+    { name: z.string().describe('The name of the memory to delete') },
+    async ({ name }) => {
+      try {
+        const deleted = await MemoryModel.deleteMemory(name);
+
+        if (!deleted) {
+          return {
+            content: [
+              {
+                type: 'text',
+                text: `Memory "${name}" not found.`,
+              },
+            ],
+          };
+        }
+
+        // Emit WebSocket event for memory update
+        const memories = await MemoryModel.getAllMemories();
+        websocketService.broadcast({
+          type: 'memory-updated',
+          payload: { memories },
+        });
+
+        return {
+          content: [
+            {
+              type: 'text',
+              text: `Memory "${name}" has been deleted.`,
+            },
+          ],
+        };
+      } catch (error) {
+        return {
+          content: [
+            {
+              type: 'text',
+              text: `Error deleting memory: ${error instanceof Error ? error.message : 'Unknown error'}`,
+            },
+          ],
+        };
+      }
+    }
+  );
+
+  archestraMcpServer.tool(
+    'search_mcp_servers',
+    'Search for MCP servers in the catalog',
     {
-      title: 'Uninstall MCP Server',
-      description: 'Uninstall an MCP server',
-      inputSchema: { id: z.string().describe('The id of the MCP server to uninstall') },
+      query: z.string().optional().describe('Search query to find specific MCP servers'),
+      category: z.string().optional().describe('Filter by category (e.g., "ai", "data", "productivity")'),
+      limit: z.number().int().positive().default(10).optional().describe('Number of results to return'),
     },
-    async ({ id }, extra) => {
+    async ({ query, category, limit }) => {
       try {
-        await McpServerModel.uninstallMcpServer(id);
+        // Search the catalog
+        const catalogUrl = process.env.ARCHESTRA_CATALOG_URL || 'https://www.archestra.ai/mcp-catalog/api';
```

Comment:
I see this config, `const catalogUrl = process.env.ARCHESTRA_CATALOG_URL || 'https://www.archestra.ai/mcp-catalog/api'`, is defined in two spots in `src/backend`, lets add it to `desktop_app/src/backend/config.ts` and consume it from there